contrib/init/upstart/docker.conf

```diff
@@ -58,7 +58,7 @@ post-start script
 	if ! printf "%s" "$DOCKER_OPTS" | grep -qE -e '-H|--host'; then
 		DOCKER_SOCKET=/var/run/docker.sock
 	else
-		DOCKER_SOCKET=$(printf "%s" "$DOCKER_OPTS" | grep -oP -e '(-H|--host)\W*unix://\K(\S+)')
+		DOCKER_SOCKET=$(printf "%s" "$DOCKER_OPTS" | grep -oP -e '(-H|--host)\W*unix://\K(\S+)' | sed 1q)
```

Comment:
Wouldn't it be slightly better to just add `-m 1` to `grep`'s arguments rather than shelling out to `sed`?